### DISCUSSION TITLE
providing family access | multiple accounts | separate providers (yt)

### DISCUSSION THREAD

**Discussion by Symbiot78** [music-assistant/support#2128] [Feature requests and ideas]
Hi
what's the best way to give your family/roommates etc. access to MA and keep separate the providers available.
specifically I have a family sub for YT music. But adding 2 user accounts just merges everything.
I'd like to give my family access to their individual stuff and let them keep their searches etc. to themselves.

I can't seem to find a way to do that.

I installed MA in a docker. But with network host setting - so I can't even run multiple MA's (which in it self is a bit overkill). I have the 2nd MA finding the speakers etc. but there are other issues because ports are already taken by the original MA installation (1400 for instance). which makes it impossible to add a 2nd instance to HA.
So only way I see right now is a separate VM to get access to the ports needed. And that's even more overkill.

What I'd like to provide is a dashboard for my family where each member can see his/her playlists, a search option etc. and keep stuff separate.

Possible?

**Comment by OzGav** [comment]
Not possible as MA is not designed that way.

**Reply by marcelveldt** [reply]
How most folks are doing this is create a dashboard in HA with just the buttons needed for the family members. That is how I also handle it for my kids. 

Splitting it all up into multiple user accounts is probably never going to happen - that is not how MA is designed.

**Reply by ItsHarper** [reply]
How does that help keep people's music collections separate from each other?

**Reply by OzGav** [reply]
I expect Marcel has buttons which activate a script to play certain albums or books etc. In answer to your question MA currently combines all sources into one library. Your option at the moment, if I understand your question, is to run multiple instances of MA.

**Reply by Hedda** [reply]
How would you handle seperate music libraries for differen  users accessing Music Assistant via a Music Assistant client mobile app?

* https://github.com/music-assistant/kmp-client-app

Never say never! ;)

**Reply by tcurdt** [reply]
> How would you handle seperate music libraries for differen users

Well, just like browsers do: select a profile.
In the profile you can configure access to the libraries.

**Comment by Symbiot78** [comment]
hi @marcelveldt  - thanks for this.
Any chance you could share a screenshot of the way you've set it up? or an example?
I struggle to come up with ideas to solve it and visual is always better :)

**Reply by loryanstrant** [reply]
I too would like to see a screenshot of this or how it's been achieved.
My use case is that in daughter #2's room she has a dedicated HA tablet mounted on the wall. The Spotify app is not as touch friendly as you'd like on a wall, whereas MA is.
My desired state is that when she loads MA on that tablet, it loads her Spotify account and plays only on her speaker.

The only way I'm thinking this can be achieved is for me to set up a separate MA instance in Docker that is only connected to her Spotify account and her speaker and embed that in a dashboard on her device.

**Reply by OzGav** [reply]
Yes a second instance would be required. You will find the actions support multiple instances.

**Reply by loryanstrant** [reply]
Yep, got it working, kinda.
Would be good if I didn't have to shut down the first MA instance so I could fire up the second instance to change the web port, but that was a small hassle.

Main thing is I can't get any speakers to show up in the second instance. None of the Sonos or DLNA speakers are appearing.
In the end, I added the second instance to HA and then added a speaker from there.

**Reply by TRusselo** [reply]
you can run 2 or more instances at the same time if you change the ports on the other instances

**Comment by skyw33** [comment]
The other use case is when you want to limit explicit songs when searching on a specific device. Are there techniques for doing that?

**Reply by OzGav** [reply]
Filtering explicit is a feature on the backlog

**Comment by thombec** [comment]
I have the same issue/problem -> two daughters and my wife (and me). Therefore, I need the option to create four different user accounts. I don't understand the idea of ​​the dashboards – could you please explain this in more detail? I haven't been able to find a solution for this problem yet (separate playlists, separate speakers, etc.).

**Comment by Hedda** [comment]
I also like to add that can't wait for Music Assistant to someday support multiple user profiles so could have multiple users with different libraries and playlists exposed so that can track different users playback and hide stuff per ago or user.

I have not seen this anyone stating that this is planned but I assume that multiple user profiles is on some roadmap? 

I would very much like to be able to create separate user profiles for each family members (and service users) so that they can manage their own separate playlists and favorites as well as not expose their playback to others in the family.

Currently, the interface has no authentication and doesn't integrate with Home Assistant users profiles to take advantage of its user accounts either.

Note that I am not asking that each users should use their own accounts to Music Providers, instead only want different users in Music Assistant.

**Reply by madcar86** [reply]
This is exactly what I'm looking for

**Comment by loryanstrant** [comment]
I think to that point, there's actually a potential financial benefit if MA supported multiple users - in that families could avoid having to pay for family accounts as MA would handle each person's preferences and favourites.
Especially as the prices are getting ridiculous and you're effectively locked in. 
Drizzle some AI in there and you've got personalised recommendations.
(I know it's a lot more complicated than I make out.)

It would however impact the ability to play audio simultaneously as you're using the one account, and I'm sure the streamers would catch wind of it and move to somehow block the functionality.

But it's nice to dream...

**Comment by Volkor3-16** [comment]
My current problem is I have ListenBrainz/Last.FM Scrobbling set up on music-assistant. Works perfectly... If I don't share the server with everyone else in the house, and mess up my suggested feed on them.

Multi-Users would make this trivial. Setting up a instance for every person seems like a messy workaround.

**Comment by TRusselo** [comment]
<img width="1195" height="780" alt="image" src="https://github.com/user-attachments/assets/f4614d71-41fc-487d-b306-8f48df7a5962" />

this would just be too simple

**Comment by OzGav** [comment]
Multiple user profiles has been added to the 2.7 beta. Note there will always be one combined library in the backend but each user can have their providers filtered in the settings.

**Reply by LutzDi** [reply]
@OzGav, these are great news, thx a lot... May I please you to lead me to the right release/issue link for this new feature? Unfortunately, I was not able to find it - what is for sure a prob on my side. Thx for pointing me to the right place so I can read what you implemented in the beta... Cheers Lutz

**Reply by marcelveldt** [reply]
Watch the livestream tomorrow when we launch and present it!

**Reply by scyto** [reply]
> Multiple user profiles has been added to the 2.7 beta. Note there will always be one combined library in the backend but each user can have their providers filtered in the settings.

I assume this still means the two users would need different providers (say one user on tidal and one user on apple, no way to have two of the same provider still), as per your original response up top)?

**Reply by OzGav** [reply]
Yes you can have two of the same provider for those that support it.

**Reply by scyto** [reply]
> Yes you can have two of the same provider for those that support it.

cool, was hunting around for info about which support it, the one i am most interested in is apple music but can't see how to add that twice as an admin

**Reply by OzGav** [reply]
I just checked the code and Apple Music supports multi instance so you should be able to add one and then go back and add another one.

**Reply by Hedda** [reply]
Can you add this info to the official end-user documentation?

Specifically list which music providers support multi-instance setups?

**Reply by OzGav** [reply]
We are looking into it. I just need to make sure that it actualy works for each provider rather than theoretically!

**Reply by scyto** [reply]
> I just checked the code and Apple Music supports multi instance so you should be able to add one and then go back and add another one.

sorry for not realzing one has to run the add provider multiple times - i was looking for an option in the settings for my first apple music provider

seems to work, thanks!

<img width="1254" height="130" alt="image" src="https://github.com/user-attachments/assets/5de55da0-047d-40cd-b670-18141ebb625c" />

**Reply by LastElf** [reply]
Tidal seems happy too, cheers for the extra filtering per user.

Is there a roadmap for HA support? Looks like I can't add multiple users to give different contexts based on the logged in user/instance of the media player yet. Can work around it in the meantime, just getting expectations.

**Reply by OzGav** [reply]
There is no discussion about HA support at this time.

**Comment by Symbiot78** [comment]
reviving this a little as I've been trying to get back into MA. 
How is everyone handling youtube music for a family? 
I've added youtube music as a provider. it worked.. out of the box. Which I hadn't expected.. :-) there's always something but not this time.

However.. If I roll this out to my users ( daughters etc ) they will be using my youtube music account and not their own. Meaning their listening history will seep into my account and mess with everything. 

Only way I see this working is setting up a youtube music provider for each member of my family and having separate po token dockers running. 

how are you guys handling this?

**Reply by OzGav** [reply]
I am not sure as I dont use YTM but just try it. A quick look at the token generator and I think you might only need one.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
